File: algorithms/sorting/python/bubble_sort.py
def bubble_sort(arr):
    swapped = True
    while swapped:
        swapped = False
        for i in range(len(arr) - 1):
            if arr[i] > arr[i + 1]:
                arr[i], arr[i + 1] = arr[i + 1], arr[i]
                swapped = True


test_array = [8, 20, -2, 4, -6]  # [ -6, -2, 4, 8, 20 ]
bubble_sort(test_array)
print(test_array)

File: algorithms/sorting/javascript/bubble-sort.js
/*
Problem: Given an array of integers, sort the array
 */

function bubbleSort(arr) {
  let swapped;
  do {
    swapped = false;
    for (let i = 0; i < arr.length - 1; i++) {
      if (arr[i] > arr[i + 1]) {
        let temp = arr[i];
        arr[i] = arr[i + 1];
        arr[i + 1] = temp;
        swapped = true;
      }
    }
  } while (swapped);
}

const testArray = [8, 20, -2, 4, -6]; // [ -6, -2, 4, 8, 20 ]
bubbleSort(testArray);
console.log(testArray);

// Big-O = O(n^2)
